src/clua-machine-util.cpp

```diff
@@ -435,8 +434,9 @@ static void check_cm_access(lua_State *L, int tabidx, bool proofs, cm_access *a,
             CM_TREE_LOG2_ROOT_SIZE);
     }
     if (proofs) {
```

Comment:
Here I think what you are doing is something like this: "If the caller asked for proofs, make sure there are proofs, otherwise don't even check if there were proofs in the access". I think it should be something like this: "If the caller asked for proofs, make sure there are proofs, otherwise, let them be optional."